app/assets/javascripts/discourse/app/static/prosemirror/lib/text-manipulation.js

```diff
@@ -103,11 +103,26 @@ export default class ProsemirrorTextManipulation {
       return;
     }
 
-    const text = head + i18n(`composer.${exampleKey}`) + tail;
-    const doc = this.convertFromMarkdown(text);
+    let text;
+    if (this.view.state.selection.empty) {
+      text = head + i18n(`composer.${exampleKey}`) + tail;
+    } else {
+      // NOTE: We need to add IGNORE here otherwise the convertFromMarkdown
+      // markdown parser will parse the text as a block instead of inline.
+      text =
```

Comment:
This is really not ideal and probably breaks other stuff...I just have no idea how to get prosemirror-markdown to parse as inline bbcode for e.g. `[spoiler]test[/spoiler]`, it only figures out it's inline with something before or after the spoiler tags